commands/template_pull.go

```diff
@@ -20,74 +20,42 @@ var (
 	pullDebug  bool
 )
 
-var supportedVerbs = [...]string{"pull"}
-
 func init() {
 	templatePullCmd.Flags().BoolVar(&overwrite, "overwrite", false, "Overwrite existing templates?")
 	templatePullCmd.Flags().BoolVar(&pullDebug, "debug", false, "Enable debug output")
 
-	faasCmd.AddCommand(templatePullCmd)
+	templateCmd.AddCommand(templatePullCmd)
 }
 
 // templatePullCmd allows the user to fetch a template from a repository
 var templatePullCmd = &cobra.Command{
-	Use: "template pull <repository URL>",
-	Args: func(cmd *cobra.Command, args []string) error {
-		msg := fmt.Sprintf(`Must use a supported verb for 'faas-cli template'
-Currently supported verbs: %v`, supportedVerbs)
-
-		if len(args) == 0 {
-			return fmt.Errorf(msg)
-		}
-
-		if args[0] != "pull" {
-			return fmt.Errorf(msg)
-		}
-
-		if len(args) > 1 {
-
-			// assume it is a local repo
-			if _, err := os.Stat(args[1]); err == nil {
-				return nil
-			}
-
-			var validURL = regexp.MustCompile(gitRemoteRepoRegex)
-			if !validURL.MatchString(args[1]) {
-				return fmt.Errorf("The repository URL must be a valid git repo uri")
-			}
-		}
-		return nil
-	},
-	Short: "Downloads templates from the specified github repo",
+	Use:   `pull [REPOSITORY_URL]`,
+	Short: `Downloads templates from the specified github repo`,
```

Comment:
I don't see where the check for at least one arg went after you refactored this.  Shouldn't it now be 

```golang 
Args: cobra.MinimumNArgs(1),
```

Reply:
I have logic for checking the first argument (as url) inside the function itself, the (1) argument had to be there since the command was `template pull` and `pull` was necessary and now since we have separated those commands into two (`pull`) it needs only 1 argument arg[0] to be valid. I fixed the check in here:

```
...
func runTemplatePull(cmd *cobra.Command, args []string) error {
	repository := ""
	if len(args) > 0 {
		repository = args[0]
	}
...
```

Reply:
I handled this problem I hope that is what you meant @LucasRoesler . I really appreciate your feedback thanks!